### PR TITLE
docs(install): add setup and usage documentation

### PR DESCRIPTION
**Pull request by sdodson99**
## Changes

* Release the latest build to NPM as an alpha version
* Add documentation for:
  * Installation
  * Setup
  * Quick Start
  * Reference
* Integrate with [Moji Message](https://emoji.seandodson.com) 🎉

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -52,7 +152,47 @@ However, this feature requires admins to manually verify comments. We recommend
 
 We plan to iterate on (or remove, depending on feedback!) this feature in the future.
 
-## Development
+## Reference
```
Comment (sdodson99) [inline README.md:155]: This is somewhat incomplete (missing types, super in-depth, etc.), but I think it captures the key interface of the package. I don't think humans should be hand-writing an API reference section 😂 In the future, we could likely generate this from JS docs.

File: README.md [hunk 2]
```diff
@@ -29,6 +41,94 @@ The rules above:
 - Allow reading all comments
 - Allow writing _unverified_ comments, ensuring that all comments go through verification
 
+## Quick Start
```
Comment (sdodson99) [inline README.md:44]: This quick start is kind of big and monolothic... might be able to break this down or simplify it in the future.

File: README.md [hunk 3]
```diff
@@ -52,7 +151,47 @@ However, this feature requires admins to manually verify comments. We recommend
...
+
+| Field        | Type                                  | Description                                                                               |
+| ------------ | ------------------------------------- | ----------------------------------------------------------------------------------------- |
+| `addComment` | `(newComment: NewComment) => Promise` | Function to add a new comment to a comment section. **Throws if adding a comment fails.** |
+| `loading`    | `boolean`                             | The loading state for adding a comment.                                                   |
+| `error`      | `Error`                               | The error if adding a comment fails.                                                      |
+
+## Contributing
+
+Please create a new issue if you have any questions, problems, or suggestions. Feel free to open a pull request if you have a feature or fix you want to contribute to the project.
```
Comment (edodso2) [inline README.md:194]: This sentence stood out to me as a little choppy. Not sure why. What about:

Feel free to create a new issue for any questions, problems, or suggestions, or open a pull request to contribute a feature or fix to the project.

File: README.md (hunk 1 above)
Comment (edodso2) [reply README.md:154]: Agreed. I think it more than sufficient for now.

File: README.md (hunk 2 above)
Comment (edodso2) [reply README.md:44]: I actually really like the quick start! I keep imagining it in an interactive style doc where the steps are on the left and code on the right and user is scrolling. Kind of how stripe does it here: https://stripe.com/docs/checkout/quickstart

The only thing I would suggest is a quick intro before jumping into the code to explain the approach with the hooks and all. Just to prime the reader so they can better follow the code. Maybe something like:

Simplify your development process with Headless Comments! Just wrap your comments section with the `FirebaseCommentsProvider` component and utilize the provided hooks, `useAddComment` and `useComments`. This allows seamless persistence and display of comments from your Firestore database, while you focus on building your comments UI! Check out the detailed example below:

File: README.md [hunk 4]
```diff
@@ -8,9 +8,21 @@ All the infrastructure you need to add a highly customizable comment section to
...
+
+Via NPM:
+
+```
+npm i @headless-comments/react firebase
+```
+
 ## Setup
 
-1. Configure Firestore security rules.
```
Comment (edodso2) [inline README.md:13]: Intro to this section? 

Headless Comments requires Firebase Firestore to persist comments. We would love to add other database options in the future!

**Review by edodso2 — APPROVED** [review]
Just have some nitpicks. Overall I think its great.

**Review by sdodson99 — COMMENTED** [review]
I think this README could still be friendlier, but this feedback helped a lot! 💯

1. Does our summary ("All the infrastructure you need...") need re-work? I put that together when I created the repository, so I didn't put much thought into it. Feel like it overhypes with "all the infrastructure", and it doesn't exactly highlight the "headless" feature. I liked that one comment you added - "while you focus on building your comments UI!" - that's really the main value proposition.
2. Do we need more emojis to build hype? 🚀

**Comment by edodso2** [comment]
> I think this README could still be friendlier, but this feedback helped a lot! 💯
> 
>     1. Does our summary ("All the infrastructure you need...") need re-work? I put that together when I created the repository, so I didn't put much thought into it. Feel like it overhypes with "all the infrastructure", and it doesn't exactly highlight the "headless" feature. I liked that one comment you added - "while you focus on building your comments UI!" - that's really the main value proposition.
> 
>     2. Do we need more emojis to build hype? 🚀

1. I agree.  Some ideas we can build on: 
"The persistence layer for your comments user interface."
"Take full control of your comments user interface, and let Headless Comments do the rest."

We might also want to just move the sentence from the Quick Start to the top of the page:
"This allows seamless persistence and display of comments from your Firestore database, while you focus on building your comments UI!"

And adjust it:
"Headless Comments allows seamless persistence and display of comments from your database, while you focus on building your comments UI!"

2. 🙌

**Comment by sdodson99** [comment]
> And adjust it: Take full control of your comments UI ✊
> 
> Headless Comments allows seamless persistence and display of comments from your database, while you focus on building your comments UI! 🚀
> 
> Not completely satisfied with any of these 🤔

These are good ideas, went with somewhat of a combination of both.

**Pull request merged**